cpp/build-support/lint_cpp_cli.py

```diff
@@ -77,6 +77,7 @@ def lint_file(path):
 
 
 EXCLUSIONS = _paths('''\
+    arrow/acero/concurrent_queue.h
```

Comment:
`concurrent_queue.h` relies on `<mutex>` which fails the linter. LMK if there's a workaround aside from this.

Reply:
Went for option 2